frappe/core/doctype/file/file.py

```diff
@@ -690,6 +689,15 @@ def save_file_on_filesystem(self):
 
 		return {"file_name": os.path.basename(fpath), "file_url": self.file_url}
 
+	def make_file_url(self):
+		file_url = None
+		safe_file_name = re.sub(r"[/\\%?#]", "_", self.file_name)
+		if self.is_private:
+			file_url = f"/private/files/{safe_file_name}"
+		else:
+			file_url = f"/files/{safe_file_name}"
```

Comment:
Avoid copying these, modify `save_file_on_filesystem` to use it. 

Rest looks fine.